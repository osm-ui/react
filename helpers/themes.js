import _ from 'lodash';
import { lighten } from 'polished';


export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    // const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.white,
        controlColor: color1,
        hoverControlColor: colors.white,

        column: {
            nav: {
                color: colors.white,
                backgroundColor: lighten(0.1, color2),
                hoverColor: colors.white,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.white,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.white,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.white,
                    borderColor: colors.white,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
    const color1 = colors[`${colorVariant}1`];
    const color2 = colors[`${colorVariant}2`];
    const color3 = colors[`${colorVariant}3`];
    const color4 = colors[`${colorVariant}4`];
    const color5 = colors[`${colorVariant}5`];

    return _.merge({}, baseConfig, {
        color: colors.anthracite1,
        backgroundColor: color2,
        borderColor: color1,
        loaderColor: colors.anthracite1,
        controlColor: color4,
        hoverControlColor: colors.anthracite1,

        column: {
            nav: {
                color: colors.anthracite1,
                backgroundColor: lighten(0.2, color2),
                hoverColor: colors.anthracite1,
                hoverBackgroundColor: color3,
            },
        },
        toolbar: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        titlebar: {
            color: colors.anthracite1,
            backgroundColor: color2,
            borderColor: color2,

            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color2,
                hoverBackgroundColor: color3,
                hoverBorderColor: color3,
                focusBackgroundColor: color3,
                focusBorderColor: color3,
                activeBackgroundColor: color3,
                activeBorderColor: color3,
            },
        },
        form: {
            button: {
                color: colors.anthracite1,
                backgroundColor: color2,
                borderColor: color5,
                hoverBackgroundColor: color3,
                hoverBorderColor: color5,
                focusBackgroundColor: color3,
                focusBorderColor: color5,
                activeBackgroundColor: color3,
                activeBorderColor: color5,

                primary: {
                    color: color2,
                    backgroundColor: colors.anthracite1,
                    borderColor: colors.anthracite1,
                    hoverBackgroundColor: colors.lightGray5,
                    hoverBorderColor: colors.lightGray5,
                    focusBackgroundColor: colors.lightGray5,
                    focusBorderColor: colors.lightGray5,
                    activeBackgroundColor: colors.lightGray5,
                    activeBorderColor: colors.lightGray5,
                },
            },
        },
    });
}


export default {
    buildDarkThemeConfig,
    buildLightThemeConfig,
};
