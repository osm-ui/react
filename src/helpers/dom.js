export function isOverflowYHidden(element) {
  return window.getComputedStyle(element).overflowY !== 'hidden';
}
