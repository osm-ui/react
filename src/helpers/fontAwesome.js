export function makePrefix(style) {
  return `fa${style[0]}`;
}

export function makeName(name) {
  return `fa-${name}`;
}
