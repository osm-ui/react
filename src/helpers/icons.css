@import url('https://use.fontawesome.com/releases/v5.2.0/css/all.css');
