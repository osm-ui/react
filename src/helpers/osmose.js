export function formatDeletedTags(fix, osmTags) {
  return {
    ...fix,
    del: fix.del.map(deletedTag => {
      return osmTags.find(tag => tag.k === deletedTag);
    })
  };
}

export function formatOsmTags(data) {
  return data.reduce((acc, item) => {
    return {
      ...acc,
      [item.k]: item.v
    };
  }, {});
}

export function fixOsmTags(osmTags, fix) {
  const newOsm = {
    ...formatOsmTags(osmTags),
    ...formatOsmTags(fix.add),
    ...formatOsmTags(fix.mod)
  };

  fix.del.forEach(tag => Reflect.deleteProperty(newOsm, tag));

  return newOsm;
}
