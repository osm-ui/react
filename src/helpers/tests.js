import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import DefaultTheme from 'themes/Default';

export const snapshotWithElementChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>
        <div />
      </TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};

export const snapshotWithTextChildren = (TestedElement, props) => {
  const result = shallow(
    <DefaultTheme>
      <TestedElement {...props}>This is a text</TestedElement>
    </DefaultTheme>
  );

  expect(result.childAt(0)).toMatchSnapshot();
};
