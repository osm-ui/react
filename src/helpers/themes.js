import _ from 'lodash';
import { lighten, darken } from 'polished';
import { injectGlobal } from 'styled-components';

export function buildDarkThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  // const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];

  /* eslint-disable no-unused-expressions */
  injectGlobal`
    .osm-ui-react-marker-${colorVariant} {
      color: ${colors.white};

      #colorized, #colorized * {
        fill: ${color2} !important;
      }
    }
  `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: colors.white,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: colors.white,
    controlColor: color1,
    hoverControlColor: colors.white,

    sidebar: {
      nav: {
        color: colors.white,
        backgroundColor: lighten(0.1, color2),
        hoverColor: colors.white,
        hoverBackgroundColor: color3
      }
    },
    toolbar: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: colors.white,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: colors.white,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: colors.white,
          borderColor: colors.white,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export function buildLightThemeConfig(baseConfig, colors, colorVariant) {
  const color1 = colors[`${colorVariant}1`];
  const color2 = colors[`${colorVariant}2`];
  const color3 = colors[`${colorVariant}3`];
  const color4 = colors[`${colorVariant}4`];
  const color5 = colors[`${colorVariant}5`];
  const darkColor1 = darken(0.27, color1);

  /* eslint-disable no-unused-expressions */
  injectGlobal`
        .osm-ui-react-marker-${colorVariant} {
            color: ${darkColor1};

            #colorized, #colorized * {
                fill: ${color2} !important;
            }
        }
    `;
  /* eslint-enable */

  return _.merge({}, baseConfig, {
    color: darkColor1,
    backgroundColor: color2,
    borderColor: color1,
    loaderColor: darkColor1,
    controlColor: color4,
    hoverControlColor: darkColor1,

    sidebar: {
      nav: {
        color: darkColor1,
        backgroundColor: color3,
        hoverColor: darkColor1,
        hoverBackgroundColor: lighten(0.14, color3)
      }
    },
    toolbar: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    titlebar: {
      color: darkColor1,
      backgroundColor: color2,
      borderColor: color2,

      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color2,
        hoverBackgroundColor: color3,
        hoverBorderColor: color3,
        focusBackgroundColor: color3,
        focusBorderColor: color3,
        activeBackgroundColor: color3,
        activeBorderColor: color3
      }
    },
    form: {
      button: {
        color: darkColor1,
        backgroundColor: color2,
        borderColor: color5,
        hoverBackgroundColor: color3,
        hoverBorderColor: color5,
        focusBackgroundColor: color3,
        focusBorderColor: color5,
        activeBackgroundColor: color3,
        activeBorderColor: color5,

        primary: {
          color: color2,
          backgroundColor: darkColor1,
          borderColor: darkColor1,
          hoverBackgroundColor: colors.lightGray5,
          hoverBorderColor: colors.lightGray5,
          focusBackgroundColor: colors.lightGray5,
          focusBorderColor: colors.lightGray5,
          activeBackgroundColor: colors.lightGray5,
          activeBorderColor: colors.lightGray5
        }
      }
    }
  });
}

export default {
  buildDarkThemeConfig,
  buildLightThemeConfig
};
