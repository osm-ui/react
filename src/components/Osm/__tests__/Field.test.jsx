import React from 'react';
import 'jest-styled-components';
import { snapshotWithElementChildren } from 'helpers/tests';
import { Editor } from 'index';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Editor.Field, { tag: 'tag' }));
});
