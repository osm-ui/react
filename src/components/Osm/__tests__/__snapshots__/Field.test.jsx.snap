// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with an element children 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif;
}

.c3.btn-default {
  font-weight: 500;
  color: #3b3f45;
  background-color: #ffffff;
  border-color: #c9cacc;
}

.c3.btn-default:hover {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c3.btn-default:focus {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c3.btn-default:active,
.c3.btn-default .active {
  color: #3b3f45;
  background-color: #eceeef;
  border-color: #c9cacc;
}

.c3.btn-primary {
  font-weight: 500;
  color: #ffffff;
  background-color: #2196f3;
  border-color: #186fc5;
}

.c3.btn-primary:hover {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c3.btn-primary:focus {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c3.btn-primary:active,
.c3.btn-primary .active {
  color: #ffffff;
  background-color: #71bcf7;
  border-color: #2196f3;
}

.c3.btn-info {
  font-weight: 500;
  color: #000a0a;
  background-color: #00c2b8;
  border-color: #00948d;
}

.c3.btn-info:hover {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c3.btn-info:focus {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c3.btn-info:active,
.c3.btn-info .active {
  color: #000a0a;
  background-color: #5cd7d1;
  border-color: #00c2b8;
}

.c3.btn-success {
  font-weight: 500;
  color: #ffffff;
  background-color: #4caf50;
  border-color: #348538;
}

.c3.btn-success:hover {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c3.btn-success:focus {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c3.btn-success:active,
.c3.btn-success .active {
  color: #ffffff;
  background-color: #7cc47f;
  border-color: #4caf50;
}

.c3.btn-warning {
  font-weight: 500;
  color: #543500;
  background-color: #ffa700;
  border-color: #de8c00;
}

.c3.btn-warning:hover {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c3.btn-warning:focus {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c3.btn-warning:active,
.c3.btn-warning .active {
  color: #543500;
  background-color: #ffd659;
  border-color: #ffa700;
}

.c3.btn-danger {
  font-weight: 500;
  color: #ffffff;
  background-color: #f44336;
  border-color: #c92d2d;
}

.c3.btn-danger:hover {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c3.btn-danger:focus {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c3.btn-danger:active,
.c3.btn-danger .active {
  color: #ffffff;
  background-color: #f8877f;
  border-color: #f44336;
}

.c3.btn-link {
  font-weight: 400;
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c3.btn-link:hover {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c3.btn-link:focus {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c3.btn-link:active,
.c3.btn-link .active {
  color: #2196f3;
  background-color: transparent;
  border-color: transparent;
}

.c3.btn {
  border-style: solid;
  border-width: 2px;
  border-radius: 4px;
}

.c3.btn-xs {
  font-size: 1.2rem;
  padding-left: 1.4rem;
  padding-right: 1.4rem;
}

.c3.btn-sm {
  padding-left: 1.6rem;
  padding-right: 1.6rem;
}

.c3.btn-md {
  padding-left: 2rem;
  padding-right: 2rem;
}

.c3.btn-lg {
  padding-left: 2.4rem;
  padding-right: 2.4rem;
}

.c3.shape-round.btn-xs {
  border-radius: 1.25rem;
}

.c3.shape-round.btn-sm {
  border-radius: 1.7rem;
}

.c3.shape-round.btn-md {
  border-radius: 2.1rem;
}

.c3.shape-round.btn-lg {
  border-radius: 2.45rem;
}

.c5 {
  opacity: 1;
  background-color: #ffffff;
  border-color: #c9cacc;
  border-width: 2px;
  border-radius: 0;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c5.form-control {
  height: 100%;
}

.c5:focus,
.c5.form-control:focus {
  opacity: 1;
  background-color: #ffffff;
  border-color: #2196f3;
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
}

.c5:disabled {
  opacity: 0.6;
}

.c1 {
  margin-bottom: 2rem;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c1 .main {
  z-index: 1;
  position: relative;
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .main .c2 {
  padding-left: 1rem;
  padding-right: 1rem;
  border-radius: 4px 0 0 4px;
}

.c1 .main .c2:focus {
  outline: none;
}

.c1 .main .value {
  -webkit-flex: 1 0 10rem;
  -ms-flex: 1 0 10rem;
  flex: 1 0 10rem;
  line-height: 3rem;
  padding: 1px 0;
  padding-left: 1rem;
  border: 1px solid #c9cacc;
  border-left: none;
  border-radius: 0 4px 4px 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c1 .main .actions {
  position: absolute;
  opacity: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  right: 0;
  top: 0;
  bottom: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c1 .main .actions .c2 {
  padding: 0 1rem;
}

.c1 .main .actions .action {
  color: #ffffff;
  line-height: 3rem;
  border: 2px solid transparent;
  cursor: pointer;
}

.c1 .main .actions .del {
  background-color: #f44336;
  border-color: #c92d2d;
  border-radius: 0 4px 0 0;
}

.c1 .main .actions .mod {
  background-color: #ffa700;
  border-color: #de8c00;
  border-radius: 0;
}

.c1.add .c2,
.c1.add .value {
  border-color: #7cc47f;
}

.c1.add .c2 {
  color: #ffffff;
  border-color: #348538;
  background-color: #7cc47f;
}

.c1.mod .c2,
.c1.mod .value {
  border-color: #ffd659;
}

.c1.mod .c2 {
  color: #ffffff;
  border-color: #de8c00;
  background-color: #ffa700;
}

.c1.del .c2,
.c1.del .value {
  border-color: #f8877f;
}

.c1.del .c2 {
  color: #ffffff;
  border-color: #c92d2d;
  background-color: #f8877f;
}

.c1.missing .value {
  color: #c92d2d;
  font-style: italic;
}

.c1.inactive,
.c1.removed {
  opacity: 0.2;
}

.c1.inactive .input-container,
.c1.removed .input-container,
.c1.inactive .textarea-container,
.c1.removed .textarea-container {
  visibility: hidden;
}

.c1 .input-container,
.c1 .textarea-container {
  height: 3.4rem;
  z-index: 0;
  margin-top: -3.4rem;
  -webkit-transition: margin-top 300ms ease-in-out,height 300ms ease-in-out;
  transition: margin-top 300ms ease-in-out,height 300ms ease-in-out;
}

.c1 .c4,
.c1 .sc-VigVT {
  line-height: 1.7rem;
  border-radius: 0 0 4px 4px;
}

.c1.selected {
  opacity: 1;
}

.c1.selected .c2 {
  border-radius: 4px 0 0 0;
}

.c1.selected .input-container,
.c1.selected .textarea-container {
  margin-top: 0;
}

.c1.selected .textarea-container {
  height: 8rem;
}

.c1.selected .value {
  border-radius: 0 4px 0 0;
}

<span
  className="c0"
>
  <div
    className="missing c1"
  >
    <div
      className="main"
    >
      <button
        className="tag btn shape-square btn-default btn-sm c2 c3"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        tag
      </button>
      <div
        className="value"
      >
        A value is required
      </div>
    </div>
    <div
      className="input-container"
      style={null}
    >
      <input
        className="form-control c4 c5"
        disabled={false}
        onChange={[Function]}
        placeholder="A value is required"
        type="text"
        value=""
      />
      
    </div>
  </div>
</span>
`;
